<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>kvWin</class>
 <widget class="QDialog" name="kvWin">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>471</width>
    <height>229</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>AddKeyValue</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QFrame" name="frame">
     <property name="styleSheet">
      <string notr="true">#frame {
	border: 1px solid black;
	border-radius: 5px;
	
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0.994, y2:1, stop:0 rgba(159, 165, 213, 255), stop:1 rgba(232, 245, 200, 255));
}

#findFileBtn {
	border: none;
	border-image: url(:/imgs/openFile.png);
}
#findFileBtn:hover {
	border-image: url(:/imgs/openFile_hover.png);
}
#findFileBtn:pressed {
	border-image: url(:/imgs/openFile_pressed.png);
}

#confirmBtn {
	border: none;
	border-image: url(:/imgs/confirm.png);
}

#confirmBtn:hover {
	border-image: url(:/imgs/confirm_hover.png);
}

#confirmBtn:pressed {
	border-image: url(:/imgs/confirm_pressed.png);
}

#cancelBtn {
	border: none;
	border-image: url(:/imgs/cancel.png);
}

#cancelBtn:hover {
	border-image: url(:/imgs/cancel_hover.png);
}

#cancelBtn:pressed {
	border-image: url(:/imgs/cancel_pressed.png);
}

#nameInput {
	font: 11pt &quot;Source Code Pro&quot;;
	padding: 2px;
	border: 2px solid #95a5a6;
	background-color: rgba(255, 255, 255, 0);
	border-radius: 3px;
}

#nameInput:hover {
	border-color: #7f8c8d;
}

#nameInput:focus {
	border-color: #bdc3c7;
}

#pathInput {
	padding: 2px;
	font: 11pt &quot;Source Code Pro&quot;;
	border: 2px solid #95a5a6;
	background-color: rgba(255, 255, 255, 0);
	border-radius: 3px;
}

#pathInput:hover {
	border-color: #7f8c8d;
}

#pathInput:focus {
	border-color: #bdc3c7;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QPushButton" name="cancelBtn">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>180</y>
        <width>41</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="nameLabel">
      <property name="geometry">
       <rect>
        <x>37</x>
        <y>59</y>
        <width>111</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>13</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>variable name</string>
      </property>
     </widget>
     <widget class="QLabel" name="valueLabel">
      <property name="geometry">
       <rect>
        <x>37</x>
        <y>120</y>
        <width>111</width>
        <height>40</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>13</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>variable value </string>
      </property>
     </widget>
     <widget class="QPushButton" name="findFileBtn">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>120</y>
        <width>40</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLineEdit" name="nameInput">
      <property name="geometry">
       <rect>
        <x>167</x>
        <y>60</y>
        <width>241</width>
        <height>40</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>25</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLineEdit" name="pathInput">
      <property name="geometry">
       <rect>
        <x>167</x>
        <y>120</y>
        <width>241</width>
        <height>40</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>25</height>
       </size>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="confirmBtn">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>170</y>
        <width>60</width>
        <height>60</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="title">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>411</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string/>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="keyWarningLabel">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>100</y>
        <width>235</width>
        <height>16</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <family>Times New Roman</family>
       </font>
      </property>
      <property name="text">
       <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600; color:#ff0000;&quot;&gt;the variable is already existing!&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
      <property name="textFormat">
       <enum>Qt::AutoText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cancelBtn</sender>
   <signal>clicked()</signal>
   <receiver>kvWin</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>342</x>
     <y>150</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>180</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>findFileBtn</sender>
   <signal>clicked()</signal>
   <receiver>kvWin</receiver>
   <slot>openFileDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>431</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>472</x>
     <y>115</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>confirmBtn</sender>
   <signal>clicked()</signal>
   <receiver>kvWin</receiver>
   <slot>confirm()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>225</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>168</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameInput</sender>
   <signal>textChanged(QString)</signal>
   <receiver>keyWarningLabel</receiver>
   <slot>hide()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>237</x>
     <y>77</y>
    </hint>
    <hint type="destinationlabel">
     <x>254</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameInput</sender>
   <signal>editingFinished()</signal>
   <receiver>kvWin</receiver>
   <slot>checkVariable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>206</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>-11</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFileDialog()</slot>
  <slot>confirm()</slot>
  <slot>checkVariable()</slot>
 </slots>
</ui>
